extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -61,6 +64,76 @@ export async function promptImportInternetDatabase(
 
 }
 
+/**
+ * Prompts a user to fetch a database from GitHub.
+ * User enters a GitHub repository and then the user is asked which language
+ * to download (if there is more than one)
+ *
+ * @param databaseManager the DatabaseManager
+ * @param storagePath where to store the unzipped database.
+ */
+export async function promptImportGithubDatabase(
+  databaseManager: DatabaseManager,
+  storagePath: string,
+  credentials: Credentials,
+  progress: ProgressCallback,
+  token: CancellationToken,
+  cli?: CodeQLCliServer
+): Promise<DatabaseItem | undefined> {
+  progress({
+    message: 'Choose repository',
+    step: 1,
+    maxStep: 2
+  });
+  const githubRepo = await window.showInputBox({
+    title: 'Enter a GitHub repository in the format <owner>/<repo> (e.g. github/codeql)',
+    placeHolder: '<owner>/<repo>',
+    ignoreFocusOut: true,
+  });
+  if (!githubRepo) {
+    return;
+  }
+
+  if (!REPO_REGEX.test(githubRepo)) {
+    throw new Error(`Invalid GitHub repository: ${githubRepo}`);
+  }
+
+  const databaseUrl = await convertGithubNwoToDatabaseUrl(githubRepo, credentials, progress);
+  if (databaseUrl) {
+    const octokit = await credentials.getOctokit();
+    /** 
+     * A token object returned by `octokit.auth()`. It's undocumented, but looks something like this:
+     * {
+     *   token: 'xxxx',
+     *   tokenType: 'oauth',
+     *   type: 'token',
+     * }
+     * We only need the actual token string.
+     */
+    const octokitToken = await octokit.auth() as { token: string };
```

Comment:
Minor: Could this be `(await octokit.auth())?.token`, and then you don't need to use `.token` everywhere below?

Reply:
Makes sense! I still had to keep `as { token: string }`, since `await octokit.auth()` is of type "unknown" otherwise 🤷🏽